app/src/main/java/org/mozilla/focus/browser/FirefoxProgressBar.kt

```diff
@@ -26,23 +25,34 @@ private const val HIDE_ANIMATION_DURATION_MILLIS = 250L
 private val HIDE_AFTER_MILLIS = TimeUnit.SECONDS.toMillis(3)
 
 class FirefoxProgressBar @JvmOverloads constructor(
-        context: Context, attrs: AttributeSet? = null, defStyle: Int = 0)
-    : LinearLayout(context, attrs, defStyle) {
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : LinearLayout(context, attrs, defStyle), Session.Observer {
 
     private val hideHandler = FirefoxProgressBarHideHandler(this)
+    private var session: Session? = null
```

Comment:
`lateinit var`? A progress bar shouldn't exist without a session so we should assert that state.

Reply:
👍